llvm/include/llvm/MC/CAS/MCCASObjectV1.h

```diff
@@ -358,7 +358,7 @@ class MCCASBuilder {
   Error finalizeGroup();
 
   void startSection(const MCSection *Sec);
-  Error finalizeSection();
+  template <typename SectionRefTy = SectionRef> Error finalizeSection();
```

Comment:
@cachemeifyoucan this commit contains all the changes I needed (for now?) to add a new CAS object type for the debug info section

Reply:
The new section type looks fine to me. Can't understand how you extract abbr field by glimpsing through, some comments or explanation in commit message is appreciated.

Also you can run the benchmark to see the gain in llvm-project builds.

Reply:
>  Can't understand how you extract abbr field by glimpsing through, some comments or explanation in commit message is appreciated.

The core functionality for extracting abbreviation offsets was done in previous PRs, inside the function  `getAndSetDebugAbbrevOffsetAndSkip`.

The first commit slightly changes it to overwrite offsets. I'll update that particular message to emphasize the point that we are not changing _how_ the offset is found